### ISSUE TITLE
Generating predictable model classes in strawberry shake

### ISSUE THREAD

**Issue by jarlef** [ChilliCream/graphql-platform#6873]
### Product

Strawberry Shake

### Is your feature request related to a problem?

A recent issue ive encounter is that the generated client in Strawberry Shake is not consistent when using fragments

E.g the following graphl queries agains a schema.

```graphql
fragment MySchool on School {
    id
    name
}

fragment MyStudent on Student {
    id
    name
    age
    school {
        ...MySchool
    }
}

query GetStudentById(id: Int!) {
    student(id: $id) {
        ...MyStudent
    }

}

query GetStudents {
    students {
        ...MyStudent
    }
}
```

In this case it will generate two models for student. `GetStudentById_Student` and `GetStudents_Students_Student` both inherting from their own dedicated interfaces that in turn inherits from `IMyStudent`. But deeply nested properties like the `School` property on `GetStudents_Students_Student` and `GetStudentById_Student` gets randomly assigned a type that is either `IGetStudents_Students_Student_School` or `IGetStudentById_Student_School`. 

### The solution you'd like

Since they are both implementing the fragment interface `IMyStudent` it clearly means the school property can only be of one type and IMO that should be `IMySchool` which is the fragments interface. One could also argu that student and students properties should also use the interface `IMyStudent` instead of their own interfaces since the fragment is an intent to express a decidered contract.

**Comment by dillon-miller2-carvana** [comment]
This is causing us a huge headache anytime we need to update the schema. We have several queries and mutations that we call that share common fragments. Due to this, whenever we update the schema, there is a high likelihood that the randomly assigned type that Strawberry Shake uses to generate the classes which utilize fragments will change as well.

**Comment by ryanhill-mastery** [comment]
Are there any updates to this? Or workarounds in the meantime? This is a pretty major issue that results in some developers being able to compile our applications and some who cannot. It also results in our build pipeline "randomly" failing.

**Comment by michaelstaib** [comment]
No, focus for version 14 is mainly Hot Chocolate with minor fixes to Strawberry Shake.
For 15 we have some time budget for Strawberry Shake, so once 14 is shipped we will go over sb issues and reprioritize some for the 15 backlog.

**Comment by dasomx77** [comment]
@michaelstaib could we at least get a summary of why this is happening so the community can maybe figure out an interim way to solve it (or help?)

**Comment by dasomx77** [comment]
@ryanhill-mastery not sure if this will help you in your case, but for my scenario, I had those queries split into their own files (and I strongly suspect that the query order is different depending on the platform, leading to inconsistent names for deeply nested fragments but consistent on the same platform if that makes sense). The first query to be built automatically gets "naming" right on the nested objects leading to that inconsistency since from a mac as an example the first query to be built will be different than windows as an example. I ended up putting all my queries into the same file and now the nested objects seem consistent across platforms. Of course this probably much easier if you don't have loads of queries laying around. Hope that helps.

**Comment by waldemarsson** [comment]
@michaelstaib  Any update on this? 15 seems to be shipped and work on 16 is ongoing now

**Comment by barnacleboy** [comment]
I also find this very annoying when updating, but also when actually looking through the code. It is very irritating to see types assigned that are generated for another query - and the current queries own type is actually not used anywhere. For my project, this mostly occurs in unit tests where query objects are arranged. You are testing code that uses a mocked query result and this result is built out of classes that are not part of this query...
